lighthouse-cli/test/smokehouse/smoke-test-dfns.js

```diff
@@ -6,128 +6,157 @@
 'use strict';
 
 const path = require('path');
-const smokehouseDir = 'lighthouse-cli/test/smokehouse/';
 
-/** @type {Array<Smokehouse.TestDfn>} */
-const SMOKE_TEST_DFNS = [{
+/**
+ * @param {string} filePath
+ */
+function pathRelativeToProjectRoot(filePath) {
+  const projectRoot = path.join(__dirname, '../../../');
+  return path.relative(projectRoot, filePath);
+}
+
+/**
+ * Attempt to resolve a path relative to the smokehouse folder.
+ * If this fails, attempts to locate the path
+ * relative to the project root.
+ * @param {string} payloadPath
+ * @return {string}
+ */
+function resolveLocalOrProjectRoot(payloadPath) {
```

Comment:
This actually still sticks arounds because the smoke runner needs a path to use with the LH CLI.

Instead, maybe the smoke runner could save a temporary file to disk, and pass that path? Does that sound better - temporary file creation vs this weird paths shenanigans?

Reply:
could we leave it where it was then to avoid the unnecessary diff/blame churn?

> Instead, maybe the smoke runner could save a temporary file to disk, and pass that path? Does that sound better - temporary file creation vs this weird paths shenanigans?

you know I support this :) this was one of my fav features of #8962

Reply:
I didn't notice the file shenanigans you did there until now (I was just focused on the jest stuff), but yeah good to see this is supported by you too!

> could we leave it where it was then to avoid the unnecessary diff/blame churn?

I am bad about deleting code when I think it's no longer needed, then adding it back when I realize I need it but putting it in a different place :(